### PR TITLE
hc: expose server context to hc context

### PR DESCRIPTION
**Pull request by botengyao**
hc: expose server context to hc context

Commit Message:
Additional Description:
Risk Level: low
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: source/common/upstream/cluster_factory_impl.cc [hunk 1]
```diff
@@ -108,7 +108,7 @@ ClusterFactoryImplBase::create(const envoy::config::cluster::v3::Cluster& cluste
       new_cluster_pair.first->setHealthChecker(HealthCheckerFactory::create(
           cluster.health_checks()[0], *new_cluster_pair.first, server_context.runtime(),
           server_context.mainThreadDispatcher(), server_context.accessLogManager(),
-          context.messageValidationVisitor(), server_context.api()));
+          context.messageValidationVisitor(), server_context.api(), server_context));
```
Comment (yanavlasov) [inline source/common/upstream/cluster_factory_impl.cc:111]: It looks like all parameters that are initialized from server_context properties can be removed and just use `server_context` member variable later on. Would it be too much work to clean these up?

**Review by yanavlasov — COMMENTED** [review]
/wait-any

File: source/common/upstream/cluster_factory_impl.cc (hunk 1 above)
Comment (botengyao) [reply source/common/upstream/cluster_factory_impl.cc]: Done.

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**